docs/source/ext/database.rst

```diff
@@ -247,3 +251,91 @@ The database schema is structured as follows:
+        ...
+
+    async def load_token(self, context: StreamFlowContext, persistent_id: int):
+        ...
+
+    async def load_workflow(self, context: StreamFlowContext, persistent_id: int):
+        ...
+
+
+Implementations
+---------------
+
+====================================================================    =============================================================
+Name                                                                    Class
+====================================================================    =============================================================
+:ref:`DefaultDatabaseLoadingContext <DefaultDatabaseLoadingContext>`    streamflow.persistent.loading_context.DefaultDatabaseLoadingContext
+:ref:`WorkflowBuilder <WorkflowBuilder>`                                  streamflow.persistent.loading_context.WorkflowBuilder
+====================================================================    =============================================================
+
+DefaultDatabaseLoadingContext
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``DefaultDatabaseLoadingContext`` keeps track of all the objects already loaded in the current transaction, serving as a cache to efficiently load nested entities and prevent deadlocks when dealing with circular references.
+Furthermore, it assigns the ``persistent_id`` when an entity is added to the cache with the ``add_entity`` method.
+
+
+WorkflowBuilder
+^^^^^^^^^^^^^^^
+The ``WorkflowBuilder`` class loads the steps and ports of an existing workflow from the database and inserts them into a new workflow object, passed as an argument to the constructor.
+Between the workflows, it is possible to have some shared entities, particularly those used only in reading, for example ``deployment``` and ``target``. Instead, the entities with an internal state must be different instances, so ``steps``, ``ports`` and ``workflow``.
+This is done by loading the entity, keeping the ``persistent_id`` in the case of a shared object, or creating a new ``persistent_id`` otherwise.
+The ``WorkflowBuilder`` class extends the ``DefaultDatabaseLoadingContext`` class; it has the ``workflow``, i.e., the new ``workflow`` instance, moreover, overwrites only the methods involving the ``step``, ``port``, and ``workflow`` entities.
+Particularly, the ``add_step``, ``add_port`` and ``add_workflow`` methods do not set the ``persistent_id`` as their parent methods.
+The class has the ``workflow``, i.e., the new ``workflow`` instance.
+The ``load_workflow`` method has two behaviors based on the calling order.
+When a ``WorkflowBuilder`` instance is created, and the ``load_workflow`` method is called as the first method, it loads all the entities of the original workflow in the new one.
+Instead, if it is called after the ``load_step`` or ``load_port`` methods, it returns the ``self.workflow`` without loading other entities.
+This allows to copy the entire workflow or load only a subset.
+
+
+.. code-block:: python
```

Comment:
This code block can be removed, as the behaviour is clear from the previous description